### PR TITLE
feat: add `Dropout` to Flax UNet

### LINKED ISSUE #522: [Community] Add dropout to Flax UNet
### Describe the bug

Flax UNet currently doesn't support dropout 

### Reproduction

_No response_

### Logs

_No response_

### System Info

Main

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
# What does this PR do?

Fixes #522 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Request for Review: @patrickvonplaten

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
cc @pcuenca

File: src/diffusers/models/attention_flax.py [hunk 1]
```diff
@@ -169,6 +169,7 @@ def reshape_batch_dim_to_heads(self, tensor):
         tensor = tensor.reshape(batch_size // head_size, seq_len, dim * head_size)
         return tensor
 
+    @nn.compact
```
Comment (pcuenca) [inline src/diffusers/models/attention_flax.py:172]: There's already a `setup` method in this class, it's a bit weird (to me) that we also use `nn.compact`. Would it be feasible to add `nn.Dropout` as a module in `setup` instead?

File: src/diffusers/models/attention_flax.py [hunk 2]
```diff
@@ -409,7 +412,7 @@ def setup(self):
         self.net_2 = nn.Dense(self.dim, dtype=self.dtype)
 
     def __call__(self, hidden_states, deterministic=True):
-        hidden_states = self.net_0(hidden_states)
+        hidden_states = self.net_0(hidden_states, deterministic=deterministic)
```
Comment (pcuenca) [inline src/diffusers/models/attention_flax.py:415]: 👍

**Review by pcuenca — COMMENTED** [review]
Thanks for this! I just have a question about the use of `nn.compact` and then we can merge.

File: src/diffusers/models/attention_flax.py (hunk 1 above)
Comment (SauravMaheshkar) [reply src/diffusers/models/attention_flax.py:172]: Yup, will add them to `setup` now.

Comment (SauravMaheshkar) [reply src/diffusers/models/attention_flax.py]: Addressed in 364432b5a58e988ad5a2f186c8a876148696bc62

**Review by pcuenca — APPROVED** [review]
Awesome, many thanks!
